scripts/x-charts.exports.json

```diff
@@ -266,6 +281,7 @@
   { "name": "useDrawingArea", "kind": "Function" },
   { "name": "useGaugeState", "kind": "Function" },
   { "name": "useSvgRef", "kind": "Function" },
+  { "name": "useUtilityClasses", "kind": "Variable" },
```

Comment:
This should not be exported

Reply:
Did some cleanup, let me know what you think. Also, open to suggestions on `BarLabelComponent`. We could make it `BarLabel` only, and rename the current one to something else, open to ideas. 😅

Reply:
Naming is always an issue with those nested level of components 😅
`BarLabel` could be renamed

- `BarLabelComponent` (swiching)
- `BarLabelItem`
- `BarLabelWithContent` (we don't care it's internal)

Reply:
🙃 Types get all funky when changing names. 

Also, should owner state be accessible in `slotProps`? eg:

```tsx
<BarChart
      slotProps={{
        barLabel: {
          ownerState: {
            seriesId: '1',
            dataIndex: 0,
            color: 'red',
            isFaded: false,
            isHighlighted: false,
          },
        },
      }}
/>
```

Reply:
> should owner state be accessible in `slotProps`

I assume yes, user can pass whatever they want, but at their own risk. Why this question?

Reply:
I was checking which props they could currently pass. It feels odd that they can pass `ownerProps`.

Also, these "overridable" components from `slots` shouldn't receive it. 🤔 

Like, if we get `slots:{bar: (props) => console.log(props) }`, we get this object:
```tsx
{
    "style": {
        "y": 200,
        "x": 427.1162790697674,
        "height": 50,
        "width": 14.883720930232558
    },
    "cursor": "unset",
    "ownerState": {
        "id": "auto-generated-id-3",
        "dataIndex": 4,
        "color": "#60009B",
        "isFaded": false,
        "isHighlighted": false
    },
    "className": "MuiBarElement-root MuiBarElement-series-auto-generated-id-3"
}
```

For style it probably makes sense, since they are computed. Now, I would argue the `ownerState` properties should be spread in the props instead of inside `ownerState`, at least for overridden components, but also `slotProps`.

The user shouldn't need to know about `ownerState`

Reply:
I re-wrote the files and types, moved `BarLabel` to `BarLabelItem` and created a specific file for `BarLabel`, which applies the theme to the component. I've moved the theme there because eslint was complaining that the component name didn't match (if i left it on `BarLabel`), but I don't know if the order is correct, like. Which of `themeAugmentation` vs `slotProps` should have precedence? 😅

Reply:
Since you've introduced a component between the styled component and the slots you could pass an empty ownerState to the `useSlotProps` and pass every info as props. Then create the owner state in the `BarLabel` component